pandoc fails to build HEAD

# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is problem with specific formulae and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new

### Bug reports:

Building `pandoc` from HEAD fails on Mac OS 10.12.2 with the following:

```
[31 of 93] Compiling Text.Pandoc.Highlighting ( src/Text/Pandoc/Highlighting.hs, dist/dist-sandbox-cb3c320b/build/Text/Pandoc/Highlighting.o )

src/Text/Pandoc/Highlighting.hs:53:1: error:
    Failed to load interface for ‘Skylighting’
    Use -v to see a list of the files searched for.
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
pandoc-1.19.1 failed during the building phase. The exception was:
ExitFailure 1
```

#### Output of `brew gist-logs pandoc`

<https://gist.github.com/86d5b435ce15c067de6375314854e2d5>

--- a/Formula/pandoc.rb
+++ b/Formula/pandoc.rb
@@ -21,9 +21,11 @@ class Pandoc < Formula
 
   def install
     cabal_sandbox do
-      # remove for > 1.19.1; compatibility with directory 1.3
-      system "cabal", "get", "pandoc"
-      mv "pandoc-1.19.1/pandoc.cabal", "pandoc.cabal"
+      if build.stable?
+        # remove for > 1.19.1; compatibility with directory 1.3
+        system "cabal", "get", "pandoc"
+        mv "pandoc-1.19.1/pandoc.cabal", "pandoc.cabal"
+      end
 
       args = []
       args << "--constraint=cryptonite -support_aesni" if MacOS.version <= :lion

